File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        n = len(s)
        cnt = 0
        ans = 0
        for i in range(n):
            if s[i] == '(' : cnt+=1
            elif s[i] == ')' : cnt -= 1
            ans = max(ans,cnt)
        return ans

File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int d = 0;
        int mx = INT_MIN;
        for(auto c : s){
            if ( c == '('){
                d++;
            }
            else if ( c == ')') d--;
            mx = max(mx,d);
        }
        return mx;

    }
};